validations/src/tests.rs

```diff
@@ -3361,6 +3445,254 @@ fn block_add_vtt_but_dont_update_merkle_tree() {
+            DataRequestEligibilityClaim::default(),
+        ),
+        vec![KeyedSignature {
+            signature: Signature::default(),
+            public_key: public_key.clone(),
+        }],
+    );
+    let reveal_transaction = RevealTransaction::new(reveal_body, vec![reveal_signature]);
+
+    let reveal_body2 = RevealTransactionBody::new(dr_pointer, reveal_value, public_key2.pkh());
+    let reveal_signature2 = sign_t2(&reveal_body2);
+    let commitment2 = reveal_signature2.signature.hash();
+
+    let commit_transaction2 = CommitTransaction::new(
+        CommitTransactionBody::new(
+            dr_pointer,
+            commitment2,
+            DataRequestEligibilityClaim::default(),
+        ),
+        vec![KeyedSignature {
+            signature: Signature::default(),
+            public_key: public_key2.clone(),
+        }],
+    );
+
+    // Include CommitTransaction in DataRequestPool
+    dr_pool
+        .process_commit(&commit_transaction, &last_block_hash)
+        .unwrap();
+    dr_pool
+        .process_commit(&commit_transaction2, &last_block_hash)
+        .unwrap();
+    dr_pool.update_data_request_stages();
+
+    let mut block_hash = Hash::default();
+
+    let x = test_block_with_drpool(
+        |b| {
+            // We include a duplicate tally
+            b.txns.reveal_txns.push(reveal_transaction.clone());
+            b.txns.reveal_txns.push(reveal_transaction.clone());
+
+            b.txns.mint = MintTransaction::new(
+                b.txns.mint.epoch,
+                ValueTransferOutput {
+                    time_lock: 0,
+                    value: b.txns.mint.output.value + 100, // reveal_fee is 50*2
+                    ..b.txns.mint.output
+                },
+            );
+
+            build_merkle_tree(&mut b.block_header, &b.txns);
+
+            block_hash = b.hash();
+
+            true
+        },
+        dr_pool,
+    );
+
+    assert_eq!(
+        x.unwrap_err().downcast::<TransactionError>().unwrap(),
+        TransactionError::DuplicatedReveal {
+            pkh: reveal_transaction.body.pkh,
+            dr_pointer,
+        },
+    );
+}
+
+#[test]
+fn block_duplicated_tallies() {
+    let reveal_value = vec![0x00];
+    let (dr_pool, dr_pointer, pkh, pkh2) = dr_pool_with_dr_in_tally_stage_2_reveals(reveal_value);
+
+    // Tally value: [integer(0), integer(0)]
+    let tally_value = vec![0x82, 0x00, 0x00];
+    let vt0 = ValueTransferOutput {
+        time_lock: 0,
+        pkh,
+        value: 500,
+    };
+    let vt1 = ValueTransferOutput {
+        time_lock: 0,
+        pkh: pkh2,
+        value: 500,
+    };
+    let tally_transaction = TallyTransaction::new(dr_pointer, tally_value.clone(), vec![vt0, vt1]);
+
+    let mut block_hash = Hash::default();
+
+    let x = test_block_with_drpool(
+        |b| {
+            // We include a duplicate tally
+            b.txns.tally_txns.push(tally_transaction.clone());
+            b.txns.tally_txns.push(tally_transaction.clone());
```

Comment:
Same as with commits, I think it's possible to create a valid tally with a different hash, for example by reordering the transactions:

    let tally_transaction2 = TallyTransaction::new(dr_pointer, tally_value.clone(), vec![vt1, vt0]);